src/functional/common.py

```diff
@@ -28,6 +29,14 @@
 @dataclass(frozen=True)
 class Dimension:
     value: str
+    local: bool = dataclasses.field(default=False)
+
+    def __gt_type__(self):
+        # TODO(tehrengruber): Revisit. This coupling to the frontend is not
+        #  optimal.
+        import functional.ffront.common_types as ct
+
+        return ct.DimensionType(self)
```

Comment:
In fact we do have a pattern already in use, which avoids this coupling. Check out the `fbuiltins.FieldOffset`. It is not optimal, but this kind of wrong way around coupling is worth almost any price to avoid.